### PR TITLE
Telemetry

### PR DESCRIPTION
**Pull request by ilblackdragon**
Implements telemetry service that reports information about the node to a given URL.
Telemetry is used if this node is a validator.

Additionally now show CPU and memory usage.

(Figuring out if mergable in master or not)

Usage:
```
./target/release/near run --telemetry-url=https://explorer.nearprotocol.com/api/nodes
```
Also this is default url for testnet. Other networks don't have a default urls.

### CODE REVIEW

File: chain/telemetry/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                    .map_err(|_err| {})
+                    .map(|_response| {}),
+            );
+        }
+    }
+}
+
+/// Send telemetry event to all the endpoints.
+pub fn telemetry(telemetry: &Addr<TelemetryActor>, content: serde_json::Value) {
+    let _ = telemetry.do_send(TelemetryEvent { content });
```
Comment (bowenwang1996) [inline chain/telemetry/src/lib.rs]: Why is `let _` needed? The return type of `do_send` is void

**Comment by frol** [comment]
`transactions[].public_key` is the last array of bytes instead of base58 as far as I see.

File: near/Cargo.toml [hunk 2]
```diff
@@ -30,6 +30,7 @@ near-pool = { path = "../chain/pool" }
 near-network = { path = "../chain/network" }
 near-jsonrpc = { path = "../chain/jsonrpc" }
 near-verifier = { path = "../runtime/verifier" }
+    near-telemetry = { path = "../chain/telemetry" }
```
Comment (frol) [inline near/Cargo.toml]: Fix indentation

**Comment by frol** [comment]
This issue also resolves #1076.

File: near/src/config.rs [hunk 3]
```diff
@@ -66,6 +67,8 @@ pub const GENESIS_CONFIG_FILENAME: &str = "genesis.json";
 pub const NODE_KEY_FILE: &str = "node_key.json";
 pub const VALIDATOR_KEY_FILE: &str = "validator_key.json";
 
+const DEFAULT_TELEMETRY_URL: &str = "https://explorer.nearprotocol.com/api/nodes/";
```
Comment (frol) [inline near/src/config.rs]: Please, remove the trailing `/` from the URL (`https://explorer.nearprotocol.com/api/nodes`). Next.js handles only the URLs without the trailing `/`. :man_shrugging:

**Comment by frol** [comment]
@ilblackdragon I am missing some information about nodes that are defined in the spec.md. I am getting the following information from nearcore telemetry hearbeat:

```js
{
  account_id: '',
  bandwidth_download: 92235,
  bandwidth_upload: 616,
  cpu: 13.833992004394531,
  latest_block_hash: '6TXiJg75edP4RVNMB4yzbF4jzZVm6VEjQxhahcDDccKH',
  latest_block_height: 419675,
  memory: 95540,
  num_peers: 6,
  signature: '',
  status: '#  419675 Downloading headers 99%'
}
```

Here are the fields that I miss:

> * `node_id` - public key (base58) that is used to identify the node
> * `moniker` -- custom name of the node, can be empty
> * `ip_address` -- can be empty

In regards to the IP address, should I rely on the TCP source IP address?

**Comment by ilblackdragon** [comment]
`ip address` - can we just get this information from the request that hits the server?
`moniker` - let's use `account_id` otherwise let's just have empty
`node_id` - indeed, can add

**Pull request merged**